src/averageAges.js

```diff
@@ -55,7 +71,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child =>
+    onlyWithSon
+      ? people.some(person => person.name === child.mother
+        && child.sex === 'm')
+      : people.some(person => person.name === child.mother));
```

Comment:
```suggestion
  const children = people.filter(child =>
    onlyWithSon
      ? findMotherWithSon
      : findMother;
```

лучше выносить функции в переменные